internal/watcher/instance/nginx_config_parser.go

```diff
@@ -113,77 +113,83 @@ func (ncp *NginxConfigParser) createNginxConfigContext(
 	rootDir := filepath.Dir(instance.GetInstanceRuntime().GetConfigPath())
 
 	for _, conf := range payload.Config {
-		formatMap := make(map[string]string)
-		err := ncp.crossplaneConfigTraverse(ctx, &conf,
-			func(ctx context.Context, parent, directive *crossplane.Directive) error {
-				switch directive.Directive {
-				case "log_format":
-					formatMap = ncp.formatMap(directive)
-				case "access_log":
-					if !ncp.ignoreLog(directive.Args[0]) {
-						accessLog := ncp.accessLog(directive.Args[0], ncp.accessLogDirectiveFormat(directive),
-							formatMap)
-						nginxConfigContext.AccessLogs = append(nginxConfigContext.AccessLogs, accessLog)
-					}
-				case "error_log":
-					if !ncp.ignoreLog(directive.Args[0]) {
-						errorLog := ncp.errorLog(directive.Args[0], ncp.errorLogDirectiveLevel(directive))
-						nginxConfigContext.ErrorLogs = append(nginxConfigContext.ErrorLogs, errorLog)
-					} else {
-						slog.WarnContext(ctx, fmt.Sprintf("Currently error log outputs to %s. Log monitoring "+
-							"is disabled while applying a config; "+"log errors to file to enable error monitoring",
-							directive.Args[0]), "error_log", directive.Args[0])
-					}
-				case "root":
-					rootFiles := ncp.rootFiles(ctx, directive.Args[0])
-					nginxConfigContext.Files = append(nginxConfigContext.Files, rootFiles...)
-				case "ssl_certificate", "proxy_ssl_certificate", "ssl_client_certificate", "ssl_trusted_certificate":
-					sslCertFile := ncp.sslCert(ctx, directive.Args[0], rootDir)
-					if !ncp.isDuplicateFile(nginxConfigContext.Files, sslCertFile) {
-						nginxConfigContext.Files = append(nginxConfigContext.Files, sslCertFile)
-					}
+		if ncp.agentConfig.IsDirectoryAllowed(strings.ToLower(conf.File)) {
```

Comment:
Instead of nesting this entire block how about a quick exit if the directory is not allowed?

```suggestion
		if !ncp.agentConfig.IsDirectoryAllowed(strings.ToLower(conf.File)) {
		    continue
		}
```		
```